### PR TITLE
Store Jobs under batch/v1 instead of deprecated extensions/v1beta1

### LINKED ISSUE #10950: Store jobs under batch not under extensions
Currently Jobs are [stored](https://github.com/openshift/origin/blob/7fbb28e56d1975de73b209a46fcd5cf3eea68c62/pkg/cmd/server/kubernetes/master_config.go#L158) under `extensions` groups which is meant to be removed in the next version (1.5) of k8s (https://github.com/kubernetes/kubernetes/issues/32763). We need to:
- [ ] - ~~migrate Jobs from `extensions` to `batch`~~
- [ ] - change the cohabitation definition in `master_config.go`
- [ ] - configure the storage layer so that jobs are still stored under current location

/cc @smarterclayton

### PR DESCRIPTION
**Pull request by soltysh**
Fixes #10950. This only changes the version under jobs are stored in etcd, the location stays the same.

@deads2k @liggitt ptal

### CODE REVIEW

**Comment by soltysh** [comment]
[test]

**Comment by mfojtik** [comment]
@soltysh I read the title as "Steve Jobs under batch/v1... " :-)

File: test/integration/storage_versions_test.go [hunk 1]
```diff
@@ -23,21 +24,12 @@ import (
...
-// func TestStorageVersionsSeparated(t *testing.T) {
-// 	runStorageTest(t, "separated",
-// 		autoscaling_v1.SchemeGroupVersion,
-// 		batch_v1.SchemeGroupVersion,
-// 		extensions_v1beta1.SchemeGroupVersion,
-// 	)
-// }
-
-func TestStorageVersionsUnified(t *testing.T) {
+func TestStorageVersions(t *testing.T) {
```
Comment (liggitt) [inline test/integration/storage_versions_test.go]: do we have a test to make sure that an extensions/v1beta1 Job can be read from storage, and if updated, gets persisted back as a batch/v1 Job?

File: pkg/cmd/server/kubernetes/master_config.go [hunk 2]
```diff
@@ -184,7 +184,7 @@ func BuildDefaultAPIServer(options configapi.MasterConfig) (*apiserveroptions.Se
 		master.DefaultAPIResourceConfigSource(),
 	)*/
 	// the order here is important, it defines which version will be used for storage
-	storageFactory.AddCohabitatingResources(extensions.Resource("jobs"), batch.Resource("jobs"))
+	storageFactory.AddCohabitatingResources(batch.Resource("jobs"), extensions.Resource("jobs"))
 	storageFactory.AddCohabitatingResources(extensions.Resource("horizontalpodautoscalers"), autoscaling.Resource("horizontalpodautoscalers"))
```
Comment (liggitt) [inline pkg/cmd/server/kubernetes/master_config.go]: @DirectXMan12 should the same storage version switch be done for HPA? I assume we'll want a similar test (store as extensions/v1beta1 HPA, read/write and ensure the update was stored as autoscaling/v1 HPA)

File: test/integration/storage_versions_test.go (hunk 1 above)
Comment (soltysh) [inline test/integration/storage_versions_test.go]: Yes, down this file: https://github.com/openshift/origin/blob/b7dfdfe5ee48941d156854432738cca7d7eaac2f/test/integration/storage_versions_test.go#L148-L153

File: pkg/cmd/server/kubernetes/master_config.go (hunk 2 above)
Comment (soltysh) [inline pkg/cmd/server/kubernetes/master_config.go]: HPA doesn't have generated client for extensions/v1beta1, that's why we're missing those.

**Comment by soltysh** [comment]
Flake #11024.

[test]

**Comment by soltysh** [comment]
Flake https://github.com/openshift/origin/issues/12540

[test]

**Comment by soltysh** [comment]
Grrr.... the failures are playing on my nerves today. I've rebased the PR and waiting for fresh results. I hit https://github.com/openshift/origin/issues/12540 once again.

**Comment by soltysh** [comment]
I can't reproduce #12558 manually locally nor in AWS, trying in this PR.

**Comment by smarterclayton** [comment]
Yikes, panic on in cluster config?

**Comment by soltysh** [comment]
> Yikes, panic on in cluster config?

That's just me chasing flakes, nothing to worry about.

**Comment by soltysh** [comment]
I've cherry-picked the fix from #12582 here and will re-run tests. That will give me double check for this error :)

**Comment by soltysh** [comment]
I've  cherry-picked the new version and let the fun begin with a new test run...

**Comment by soltysh** [comment]
@liggitt anything else left here or this is good to merge?

**Comment by liggitt** [comment]
LGTM

**Comment by soltysh** [comment]
[merge]

**Pull request merged**